src/utils.rs

```diff
@@ -0,0 +1,109 @@
+//!
+//! Some API utils
+//!
+
+use rsfbclient_core::{FirebirdClient};
+use crate::{FbError, Connection, SimpleConnection, Queryable};
+use std::cmp::*;
+use std::convert::From;
+
+/// Infos about the server, database, engine...
+pub trait SystemInfos {
+
+    /// Return the current connected database name
+    fn db_name(&mut self) -> Result<String, FbError>;
+
+    /// Return the current server version
+    fn server_engine(&mut self) -> Result<EngineVersion, FbError>;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum EngineVersion {
+    V1,
+    V2,
+    V3,
+    V4,
+}
+
+impl From<EngineVersion> for u8 {
+    fn from(eg: EngineVersion) -> Self {
+        match eg {
+            EngineVersion::V1 => 1,
+            EngineVersion::V2 => 2,
+            EngineVersion::V3 => 3,
+            EngineVersion::V4 => 4
+        }
+    }
+}
```

Comment:
You can define like this:

```suggestion
#[derive(Debug, Copy, Clone)]
#[repr(u8)]
pub enum EngineVersion {
    V1 = 1,
    V2 = 2,
    V3 = 3,
    V4 = 4,
}
```

And then convert to `u8` with `EngineVersion::V2 as u8`

Reply:
Nice. I will fix